enterprise/admin/__init__.py

```diff
@@ -176,49 +172,66 @@ def get_form(self, request, obj=None, **kwargs):
         form.user = request.user
         return form
 
-    @staticmethod
-    def logo(instance):
+    def enable_dsc(self, instance):
         """
-        Instance is EnterpriseCustomer.
+        Return True if data sharing consent is enabled for EnterpriseCustomer.
+
+        Arguments:
+            instance (enterprise.models.EnterpriseCustomer): `EnterpriseCustomer` model instance
         """
-        if instance.branding_configuration:
-            return instance.branding_configuration.logo
-        return None
+        return instance.enable_data_sharing_consent
 
-    @staticmethod
-    def identity_provider(instance):
+    enable_dsc.boolean = True
+    enable_dsc.short_description = u'Enable DSC'
+
+    def has_logo(self, instance):
         """
-        Instance is EnterpriseCustomer.
+        Return True if EnterpriseCustomer has a logo.
 
-        Return identity provider name to display in enterprise customer list admin view, and if identity provider name
-        is not available then return identity provider id.
+        Arguments:
+            instance (enterprise.models.EnterpriseCustomer): `EnterpriseCustomer` model instance
         """
-        ec_idp = instance.enterprise_customer_identity_provider
-        return ec_idp.provider_name or ec_idp.provider_id if ec_idp else False
+        has_logo = False
+        if hasattr(instance, 'branding_configuration') and instance.branding_configuration.logo:
+            has_logo = True
+
+        return has_logo
 
-    def enterprise_catalog(self, instance):
+    has_logo.boolean = True
+    has_logo.short_description = u'Logo'
+
+    def has_identity_provider(self, instance):
         """
-        Enterprise catalog id with a link to catalog details page.
+        Return True if EnterpriseCustomer has related identity provider.
 
         Arguments:
             instance (enterprise.models.EnterpriseCustomer): `EnterpriseCustomer` model instance
+        """
+        identity_provider = instance.enterprise_customer_identity_provider
+
+        has_identity_provider = False
+        if identity_provider and (identity_provider.provider_id or identity_provider.provider_name):
```

Comment:
Isn't `provider_id` a required field of `EnterpriseCustomerIdentityProvider` ? if so I think we can simply do this.
```
return hasattr(instance.enterprise_customer_identity_provider)
```